lms/envs/common.py

```diff
@@ -2963,6 +2963,10 @@ def _make_locale_paths(settings):
 # visible. We default this to the legacy permission 'see_exists'.
 COURSE_ABOUT_VISIBILITY_PERMISSION = 'see_exists'
 
+# [COLARAZ_CUSTOM]
+# it will restrict anonymous users from viewing courses on index, courses and course about pages
+SHOW_COURSES_TO_ANONYMOUS_USERS = False
```

Comment:
ca we add a unit test for this setting? (there is not later :P) also please make sure to ask DevOps about this flag, so they can add this in the config var.

Reply:
Yes, I will get that variable added once you approve this PR.
Secondly, a lot of test cases need to be written for Colaraz's edx-platform, don't you think we need to have a meeting with them and get extra for writing tests?

Reply:
sad but true :) I think we should give estimates including effort for writing tests and tests should be part of the changes.